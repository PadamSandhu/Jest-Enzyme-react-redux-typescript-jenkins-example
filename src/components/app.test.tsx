import * as React from "react";
import { shallow } from "enzyme";
import App from "./App";

describe("App", () => {
    const app = shallow(<App />);
    it("renders properly", () => {
        expect(app).toMatchSnapshot();
    });
    it("Contains a connected wallet component", () => {
        // console.log(app.debug());                   // Uncheck to observe why it's Connect(Wallet) not Wallet component
        expect(app.find("Connect(Wallet)").exists()).toBe(true);
    });
});