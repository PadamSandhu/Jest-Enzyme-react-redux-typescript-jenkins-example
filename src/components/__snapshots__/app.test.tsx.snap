// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
           Loot Check 
        </h2>,
        <hr />,
        <Connect(Wallet) />,
        <Connect(Loot) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Loot Check ",
        },
        "ref": null,
        "rendered": " Loot Check ",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
             Loot Check 
          </h2>,
          <hr />,
          <Connect(Wallet) />,
          <Connect(Loot) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Loot Check ",
          },
          "ref": null,
          "rendered": " Loot Check ",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
